test/libsolidity/astPropertyTests/forLoop/isSimpleCounterLoop.sol

```diff
@@ -0,0 +1,180 @@
+type UINT is uint;
+
+function add(UINT x, UINT y) pure returns (UINT) {
+    return UINT.wrap(UINT.unwrap(x) + UINT.unwrap(y));
+}
+
+function lt(UINT x, UINT y) pure returns (bool) {
+    return UINT.unwrap(x) < UINT.unwrap(y);
+}
+
+using {lt as <, add as +} for UINT global;
+
+function g() pure returns (bool) {
+    return false;
+}
+
+function h() pure returns (uint) {
+    return 13;
+}
+
+contract C {
+    uint[] dynArray;
+    uint z = 0;
+
+    function modifyStateVarZ() public returns (uint) {
+        z = type(uint).max;
+        return z;
+    }
+
+    function f() public {
+        // Positive Cases
+        /// SimplePreIncrement: isSimpleCounterLoop
```

Comment:
I renamed the test cases so it is more clear what they check for.
I guess it would also be nice to split them into two files, one containing the positive cases and another containing the negative ones.